### PR TITLE
[react-dragula] add types

### PR DESCRIPTION
**Pull request by AdrianMrn**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x]  Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-dragula/react-dragula-tests.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import reactDragula from 'react-dragula';
```
Comment (peterblazejewicz) [inline types/react-dragula/react-dragula-tests.tsx:2]:
```suggestion
import React = require('react');
import reactDragula = require('react-dragula');
```
see my comment on `esModuleInterop`

File: types/react-dragula/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "jsx": "react",
+        "esModuleInterop": true
```
Comment (peterblazejewicz) [inline types/react-dragula/tsconfig.json:20]:
```suggestion
        "jsx": "react"
```
This is not advised to be used in DT package itself, unless there is a solid reason (e.g. 3rd party library cannot be correctly imported for tests or reference in DT package). This is for consumers of the DT package itself.
Your DT will compile and pass tests with that change (and reworked imports in tests) just fine.

**Review by peterblazejewicz — COMMENTED** [review]
@AdrianMrn it's ready to be merged, after those little corrections (if agreed, apply suggestions).

File: types/react-dragula/tsconfig.json (hunk 2 above)
Comment (AdrianMrn) [reply types/react-dragula/tsconfig.json:20]: That's great, thanks a bunch for the review and suggestions!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@AdrianMrn thx for the contribution!

**Pull request merged**